From: Arturo Buzarra <arturo.buzarra@digi.com>
Date: Tue, 21 Jan 2025 12:04:52 +0100
Subject: [PATCH] config_board: fix support for web camera with STM32MP255
 processor

This commit disables the dual camera pipeline support for USB cameras, as it is
exclusive to cameras using the DCMIPP peripheral.

Signed-off-by: Arturo Buzarra <arturo.buzarra@digi.com>
---
 .../files/resources-files/config_board.sh     | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/resources-files/config_board.sh b/resources-files/config_board.sh
index a623086..e7e8b00 100644
--- a/resources-files/config_board.sh
+++ b/resources-files/config_board.sh
@@ -15,10 +15,28 @@ STM32MP157="stm32mp157"
 STM32MP157FEV1="stm32mp157f-ev1st"
 STM32MP255="stm32mp255"
 STM32MP257="stm32mp257"
 STM32MP257FEV1="stm32mp257f-ev1st"

+function is_dcmipp_present() {
+  DCMIPP_SENSOR="NOTFOUND"
+  # ov5640 or imx335 camera can be present on csi connector
+  for video in $(find /sys/class/video4linux -name "video*" -type l);
+  do
+    if [ "$(cat $video/name)" = "dcmipp_main_capture" ]; then
+      for sub in $(find /sys/class/video4linux -name "v4l-subdev*" -type l);
+      do
+        subdev_name=$(tr -d '\0' < $sub/name | awk '{print $1}')
+        if [ "$subdev_name" = "ov5640" ] || [ "$subdev_name" = "imx335" ]; then
+            DCMIPP_SENSOR=$subdev_name
+            break;
+        fi
+      done
+    fi
+  done
+}
+
 if [[ "$FRAMEWORK" == "nbg" ]]; then
   NN_EXT=".nb"
 elif [[ "$FRAMEWORK" == "tflite" ]]; then
   NN_EXT=".tflite"
 elif [[ "$FRAMEWORK" == "onnx" ]]; then
@@ -107,10 +125,19 @@ if [[ "$COMPATIBLE" == *"$STM32MP255"* ]]; then
     OBJ_DETEC_MODEL="coco_ssd_mobilenet/ssd_mobilenet_v2_fpnlite_10_256_int8$NN_EXT"
     OBJ_DETEC_MODEL_LABEL="coco_ssd_mobilenet/labels_coco_dataset_80"
   fi
   OBJ_DETECT_DATA="coco_ssd_mobilenet/testdata/"
   OPTIONS="--dual_camera_pipeline"
+
+  is_dcmipp_present
+  if [ "$DCMIPP_SENSOR" != "NOTFOUND" ]; then
+    # DCMIPP camera
+    OPTIONS="--dual_camera_pipeline"
+  else
+    # Web camera
+    OPTIONS=""
+  fi
 fi

 if [[ "$COMPATIBLE" == *"$STM32MP257"* ]]; then
   SEMANTIC_SEGMENTATION_MODEL="deeplabv3/deeplabv3_257_int8_per_tensor$NN_EXT"
   SEMANTIC_SEGMENTATION_LABEL="deeplabv3/labels_pascalvoc"
--
2.34.1
